tests/k8s_otel/conftest.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+import os
+from pathlib import Path
+from pytest_kind import KindCluster, cluster
+
+from jina.logging.logger import JinaLogger
+from .kind_wrapper import KindClusterWrapperV2
+
+# The default version broke cni at some point. That's why we need to specify the version here.
+# This can and probably should be put in env variable actually.
+cluster.KIND_VERSION = 'v0.11.1'
+
+# TODO: Can we get jina image to build here as well?
+@pytest.fixture(scope='session' ,autouse=True)
+def build_and_load_images(k8s_cluster_v2: KindClusterWrapperV2) -> None:
+    CUR_DIR = Path(__file__).parent
+    k8s_cluster_v2.build_and_load_docker_image(str(CUR_DIR / 'test-instrumentation'), 'test-instrumentation', 'test-pip')
+    k8s_cluster_v2.load_docker_image(image_name='jinaai/jina', tag='test-pip')
+    os.environ['JINA_GATEWAY_IMAGE'] = 'jinaai/jina:test-pip'
+    yield
+    del os.environ['JINA_GATEWAY_IMAGE']
+    k8s_cluster_v2.remove_docker_image('test-instrumentation', 'test-pip')
+
+
+@pytest.fixture(scope='session')
+def k8s_cluster_v2(kind_cluster: KindCluster) -> KindClusterWrapperV2:
+    return KindClusterWrapperV2(kind_cluster, JinaLogger('kubernetes-cluster-logger'))
+
+
+@pytest.fixture(scope='session')
+def otel_test_namespace(k8s_cluster_v2: KindClusterWrapperV2) -> str:
+    """
+    Returns a namespace with the otlp-collector, jaeger and prometheus services deployed.
+
+    Service endpoints:
+        - otlp-collector:4317 - OTLP GRPC metrics endpoint
```

Comment:
You can get random ports from `jina.helper.random_port()` and export them as environment variables so the hard coded ones don't accidentally collide.

Reply:
Ah, the port forward chooses a random port on the host so shouldn't be an issue. The hard coded ports here live on the kubernetes services in which case a collision is fine; they are distinguished by service name.